pinot-spi/src/main/java/org/apache/pinot/spi/recordtransformer/enricher/RecordEnricher.java

```diff
@@ -16,25 +16,26 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.pinot.spi.recordenricher;
+package org.apache.pinot.spi.recordtransformer.enricher;
 
-import java.util.List;
 import org.apache.pinot.spi.data.readers.GenericRow;
+import org.apache.pinot.spi.recordtransformer.RecordTransformer;
 
 
 /**
- * Interface for enriching records.
- * If a column with the same name as the input column already exists in the record, it will be overwritten.
+ * Record enricher is a special {@link RecordTransformer} which is applied before other transformers to enrich the
+ * columns. If a column with the same name as the input column already exists in the record, it will be overwritten.
  */
-public interface RecordEnricher {
-  /**
-   * Returns the list of input columns required for enriching the record.
-   * This is used to make sure the required input fields are extracted.
-   */
-  List<String> getInputColumns();
+public interface RecordEnricher extends RecordTransformer {
 
   /**
    * Enriches the given record, by adding new columns to the same record.
    */
   void enrich(GenericRow record);
```

Comment:
Do we even need this enrich interface and method ? Why not go all the way and make these enrichers transformers

Reply:
Technically we don't need this interface. I kept it here because:
1. The contract for `RecordEnricher` is different from `ExpressionTransformer` as it will override existing column
2. `RecordEnricher` is designed to be pluggable

Having a separate interface makes it more clear